doc/developers-guide/accumulators.md

```diff
@@ -30,132 +30,111 @@ a) performance - if we need to do something involving inspecting a stanza or mor
 
 b) debugging - it is now very easy to produce an exact track record of a stanza
 
-c) simplified implementation of modules which inherently involve multi-stage processing (e.g. mod_amp)
+c) simplified implementation of modules which inherently involve multi-stage processing (e.g. `mod_amp`)
 
-## Implementation
+## API
 
-The accumulator is implemented in the `mongoose_acc` module, which exposes a number of methods for creating and modifying stanzas and retrieving values from them.
-Under the hood an accumulator is just a map, but it may change in the future.
+`mongoose_acc` module exports `t()` type which is the accumulator type.
 
-Initially an accumulator should be created from an `#xmlel{}` using `mongoose_acc:from_element/1` or `mongoose_acc:from_element/3`.
-It should also contain `from`, `from_jid`, `to` and `to_jid` (those are set automatically if `mongoose_acc::from_element/3` is used), `user` and `server`.
-Instantiation functions also set `timestamp` and `ref`.
-The `to` part is optional, since not every stanza is directed to a specified jid.
+### `new(new_acc_params())`
 
-Some attributes are calculated upon request and stored: these are `command`, `xmlns` and `iq_query_info`.
-If they are needed, you should first use `mongoose_acc:require/2` and then just get:
+A constructor for accumulators. `new_acc_params()` is a map with following supported keys:
 
-```
-Acc1 = mongoose_acc:require(command, Acc),
-Cmd = mongoose_acc:get(command, Acc1)
-```
+* `location` - Should be a `{Module, Function, Line}` tuple (may be constructed with `?LOCATION` macro from `mongoose.hrl`). Its format is not enforced by the acc logic but Dialyzer will most probably complain about any other type.
+* `lserver` - Nameprepped domain, which is a context for current processing.
+* `element` (optional) - If present, will be used as a source for `stanza` map.
+* `from_jid`, `to_jid` (optional) - Values used to override `from` and `to` attributes of the `element`, respectively.
 
-You can set whatever attributes you want, but treat them as read-only - an accumulator is meant
-to accumulate results and carry them forward with it, not to introduced a semi-mutable data structure.
-If you put a value to a key which already exists it will work, at least as of today, but may also produce a warning.
-Some day it may stop working, you have been warned.
+If `element` is provided, the call will fail with an exception, if it's impossible to extract both sender and recipient JID, either from stanza or `to_jid` and `from_jid`.
```

Comment:
If `element` is provided, the sender and recipient JIDs are extracted, either from the element itself, or from`to_jid` and `from_jid` parameters. The call will fail with an exception if it's not possible.
?